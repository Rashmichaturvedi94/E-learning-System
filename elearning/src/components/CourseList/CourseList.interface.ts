export interface CourseListProps {
  // TODO: add required props below
  myProp?: boolean;
}
