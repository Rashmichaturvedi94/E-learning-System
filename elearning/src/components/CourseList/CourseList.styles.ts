import styled from 'styled-components/native';
import { CourseListProps } from './CourseList.interface';

export const CourseList = styled.View<Pick<CourseListProps, 'myProp'>>`
  /* Add styles here */
`;
