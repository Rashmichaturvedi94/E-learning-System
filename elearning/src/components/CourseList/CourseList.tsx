import { CourseItem } from 'components/CourseItem';
import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native';
import { CourseListProps } from './CourseList.interface';
import { CourseList as List } from './CourseList.styles';

export const CourseList: FC<CourseListProps> = ({ data, onItemPress }) => {
  return (
    <List
      horizontal
      data={data}
      renderItem={({ item }) => {
        return (
          <TouchableOpacity
            accessible
            testID="CourseButton"
            onPress={() => {
              if (onItemPress) {
                onItemPress(item);
              }
            }}
          >
            <CourseItem item={item} />
          </TouchableOpacity>
        );
      }}
      keyExtractor={(item, index) => index.toString()}
    />
  );
};
