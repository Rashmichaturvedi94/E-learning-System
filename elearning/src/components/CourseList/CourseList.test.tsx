import React from 'react';
import { render } from '@testing-library/react-native';
import { CourseList } from './CourseList';

describe('components/CourseList', () => {
  it('should render', () => {
    const component = render(<CourseList />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
