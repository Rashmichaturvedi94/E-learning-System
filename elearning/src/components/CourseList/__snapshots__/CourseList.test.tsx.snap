// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CourseList should render 1`] = `
<RCTScrollView
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "marginTop": 8,
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
